src/protos/controls.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+/*************************************************/
+/******** Main messages that will be sent ********/
+/*************************************************/
+
+message controlConfig {
+    // The main firing trigger that requests a dart out the barrel, ASAP
+    int32 triggerSwitchPin = 1;
+    SwitchOrientation TriggerSwitchOrientation = 2;
+    // An optional switch that requests the flywheels rev up to full (or idle? partial?) speed
+    int32 optRevSwitchPin = 3;
+    SwitchOrientation optRevSwitchOrientation = 4;
+    // Which button is this?
+    int32 buttonPin = 5;
+    // This is internally routed on Detlaff; you probably don't want to touch it
+    // Is there any reason a user should want to touch this?
```

Comment:
Ack

Reply:
Wait we need batteryAdcPin back

Reply:
I thought the plan was to keep track of the hardware version, and the Detlaff could figure out which set of pins it has connected? There's really no reason to involve the user/web app in something that they can't change

Reply:
Hmm I see three options now:

Have different binaries for each hardware version which have the hardware-specific parameters hard coded. Complicates the build system but is most efficient.

Have one binary for all hardware versions, with a table stored in the Dettlaff to look up the hardware-specific parameters from the hardware version the user selects in the GUI (more flash usage but smaller protobufs)

Have one binary for all hardware versions, with the table stored in the web app back end (not accessible via the GUI). This was my original plan, the user would select the hardware version in the GUI and the web app would send Dettlaff all the hardware-specific parameters. (Less flash space but bigger protobufs. Unless the protobufs take up more flash than the above table would?)

Reply:
The parameters that would be eliminated from the protobuf by this would be the 4 pusher pins, flywheel power pin, 4 ESC pins, telemetry pin, battery adc pin, and pusher dead time

Reply:
Yeah, I would lean towards the third option of having the table on the Detlaff and we would just load the appropriate parameters based on a hardware version stored in flash. This would also be pretty easy to hook into a build system to trim down if we deprecate versions.

If the functionality / number of pins doesn't change much between versions, we could even program them in flash and load them as we boot up.

For brushed flywheel PWM, could we not hook one of the esc pins into a MOSFET? That assumes the user can find a MOSFET (or pair of MOSFETs) that work with our logic level.

Reply:
Ok, let's go with the third option then.

To do PWM you need a push/pull MOSFET driver to be able to switch the MOSFET at high enough frequency. And nerf motors draw enough current that you probably want a half bridge for better efficiency. That's not trivial for a user to set up, and off the shelf drivers like the [Pololu VNH5019](https://www.pololu.com/product/1451) or [18v25](https://www.pololu.com/product/2994) are not cheap, I don't think it's worth considering this scenario for a while.